frontend/apps/reader/modules/readerhighlight.lua

```diff
@@ -1058,6 +1096,33 @@ dbg:guard(ReaderHighlight, "onShowHighlightMenu",
             "onShowHighlightMenu must not be called with nil self.selected_text!")
     end)
 
+function ReaderHighlight:_getHighlightMenuAnchor()
+    local position = G_reader_settings:readSetting("highlight_dialog_position", "center")
+    if position == "center" then return end
+    local dialog_box = self.highlight_dialog:getContentSize()
+    local anchor_x = math.floor((self.screen_w - dialog_box.w) / 2) -- center by width
+    local screen_h = self.screen_h - Size.padding.small -- do not stick to the edge
+    local anchor_y, prefers_pop_down
+    if position == "top" then
+        anchor_y = Size.padding.small
+        prefers_pop_down = true
+    elseif position == "bottom" then
+        anchor_y = screen_h
+    else -- "gesture"
+        local text_box = self.ui.document:getWordFromPosition(self.gest_pos).sbox
+        if self.ui.paging then
+            text_box = self.view:pageToScreenTransform(self.ui.paging.current_page, text_box)
+        end
+        anchor_y = text_box.y + text_box.h + Size.padding.small -- do not stick to the box
+        if anchor_y + dialog_box.h <= screen_h then -- enough room below box with gest_pos
+            prefers_pop_down = true
+        else -- above box with gest_pos
+            anchor_y = text_box.y - Size.padding.small
```

Comment:
Shouldn't there be a `- text_box.h` here so it is fully above text_box.y ?

Reply:
https://github.com/koreader/koreader/blob/88d6613feda7d174f0d096854e4f2373597b01fe/frontend/ui/widget/container/movablecontainer.lua#L139-L140